### PR TITLE
Add io.github.quotient_im.Quaternion

### PR DESCRIPTION
**Pull request by KitsuneRal**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  Quaternion is a cross-platform desktop IM client for the Matrix protocol.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am a lead developer of the project.

This is meant to replace https://github.com/flathub/com.github.quaternion

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.quotient_im.Quaternion.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+id: com.github.quaternion
+rename-icon: quaternion
+runtime: org.kde.Platform
+runtime-version: '6.7'
```
Comment (hfiguiere) [inline io.github.quotient_im.Quaternion.yaml:4]: 6.8 is out now

File: io.github.quotient_im.Quaternion.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+modules:
+- name: libolm
+  buildsystem: cmake-ninja
+  sources:
+  - type: git
+    url: https://gitlab.matrix.org/matrix-org/olm.git
+    tag:  '3.2.15'
+  config-opts:
+  - -DBUILD_SHARED_LIBS=OFF
+  cleanup: [ /include, /lib, /share ]
```
Comment (hfiguiere) [inline io.github.quotient_im.Quaternion.yaml:27]:
```suggestion
  cleanup:
    - /include
    - /lib
    - /share    
```

please use proper YAML syntax.

Comment (hfiguiere) [reply io.github.quotient_im.Quaternion.yaml:27]: same applies below.

Comment (KitsuneRal) [reply io.github.quotient_im.Quaternion.yaml:27]: It is proper YAML. But I understood that you request to use the block syntax instead of the flow syntax.

**Comment by bbhtt** [comment]
> Looks like I cannot change app-id without editing files in the original repo... As the first step, I hoped to re-release the same tag as used last in https://github.com/flathub/com.github.quaternion, under a new app-id - but that is not possible?

You can create a patch with the changes, and add the patch here. Or use a different branch on your repo then use the commit from that branch.

File: io.github.quotient_im.Quaternion.yaml (hunk 2 above)
Comment (bbhtt) [reply io.github.quotient_im.Quaternion.yaml:27]: this should be part of a toplevel cleanup rule so that you don't need to repeat it for every module.

Comment (bbhtt) [reply io.github.quotient_im.Quaternion.yaml:27]: /lib/pkgconfig, /share/man too.

**Comment by KitsuneRal** [comment]
Given that I planned to release the next beta version, it's easier to just go forward with the beta. Is it possible to release first to the beta channel? Should be rather natural.

**Comment by hfiguiere** [comment]
I would discourage using the beta channel

**Comment by KitsuneRal** [comment]
Altogether? Suggest just throwing an undertested version on people?

**Comment by KitsuneRal** [comment]
Okay, I think I can understand the logic behind the discouragement - given that the beta channel is not advertised pretty much anywhere, few users know about it, let alone use it. Guess I could just have a PR opened in anticipation of the next release and build betas in it as I move closer to the final version.

**Comment by bbhtt** [comment]
Infra outage going on, will start the build once that is fixed.

**Comment by bbhtt** [comment]
bot, build io.github.quotient_im.Quaternion

**Comment by bbhtt** [comment]
```json
{
    "errors": [
        "appid-filename-mismatch"
    ],
    "info": [
        "appid-filename-mismatch: Appid is com.github.quaternion but Manifest filename is: io.github.quotient_im.Quaternion"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

File: io.github.quotient_im.Quaternion.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+id: com.github.quaternion
```
Comment (bbhtt) [inline io.github.quotient_im.Quaternion.yaml:1]:
```suggestion
id: io.github.quotient_im.Quaternion
```

**Comment by bbhtt** [comment]
You can trigger the build by commenting once above is fixed.

**Comment by KitsuneRal** [comment]
bot, build io.github.quotient_im.Quaternion

**Comment by hfiguiere** [comment]
too late.

**Comment by bbhtt** [comment]
It would been nice to know that before, I would have merged it to `beta` branch.

Cancelled the current build https://buildbot.flathub.org/#/builders/6/builds/159674 so that it doesn't get published to stable

If you want to publish to beta, `git checkout -b beta master` then push the branch, this will trigger a build.

Do NOT push or merge to `master` until you are ready.

**Comment by bbhtt** [comment]
Once you pushed the beta branch I can switch the default branch of the repo.

**Comment by hfiguiere** [comment]
Note that the EOL rebase WILL fail because it will redirect people to the stable branch which will not exist.

And there will be no way to tell beta user to switch to the stable branch at any point in time.

**Comment by KitsuneRal** [comment]
Well, I actually mentioned about my plan: https://github.com/flathub/flathub/pull/5777#issuecomment-2452297422 - since the repo is there (thanks), I will just open another PR on it and keep it open until I reach the quality worth releasing. Then I will merge the PR and EOL `com.github.quaternion`. Should be 1-1.5 months until then. Does that sound reasonable/feasible?

**Comment by bbhtt** [comment]
So you don't want to publish anything under the new ID `io.github.quotient_im.Quaternion` to the repos right now?

**Comment by KitsuneRal** [comment]
Correct - not to publish anything right now, but rather to work in a PR and merge it (=publish a release) in another month.

**Comment by bbhtt** [comment]
Ok, then please don't push to either master and beta branch of `io.github.quotient_im.Quaternion`. Once it is ready, merge to master -> this will publish a "stable" build.

Once that is done you can EOL rebase `master` git branch of `com.github.quaternion` to `io.github.quotient_im.Quaternion` (master branch ie. the stable repo).

**Comment by KitsuneRal** [comment]
Perfect, thanks!

**Pull request closed without merging**